src/deserialize/parse.rs

```diff
@@ -13,6 +13,14 @@ pub(crate) fn parse(
         Some((_, '(')) => read_list(chars)?,
         Some((_, '#')) => tagged_or_set(chars)?,
         Some((_, '{')) => read_map(chars)?,
+        Some((_, ';')) => {
+            let cmt = read_comment(chars)?;
+            match chars.next() {
+                Some(c) => parse(Some(c), chars)?,
+                None => cmt,
+            }
+        }
+        Some((_, s)) if s.is_whitespace() => parse(chars.next(), chars)?,
```

Comment:
I don't get why we need this extra parse? the code in general already discards whitespaces and commas.

Reply:
independent of the case with comments, if i comment that line, e.g. the following test will fail:
```
    #[test]
    fn parse_simple_map() {
        let mut edn = " {:a \"2\" :b false :c nil }".chars().enumerate();

        assert_eq!(
            parse(edn.next(), &mut edn).unwrap(),
            Edn::Map(Map::new(
                map! {":a".to_string() => Edn::Str("2".to_string()),
                ":b".to_string() => Edn::Bool(false), ":c".to_string() => Edn::Nil}
            ))
        );
    }
```
because of not handling whitespace at the top level (unless I'm missing where this is done at another level, which is entirely possible)